src/Model/Payment/AbstractPayment.php

```diff
@@ -120,10 +127,50 @@ public function isModuleActive(): bool
      * Checks if a payment method is enabled
      *
      * @return bool
+     * @throws ShopgateLibraryException
      */
     public function isEnabled(): bool
     {
-        return (bool) $this->scopeConfig->getConfigByPath(static::XML_CONFIG_ENABLED)->getValue();
+        $this->validatePaymentSettings();
+
+        return (bool)$this->scopeConfig->getConfigByPath(static::XML_CONFIG_ENABLED)->getValue();
+    }
+
+    /**
+     * Validate the payment settings
+     *
+     * @throws ShopgateLibraryException
+     */
+    private function validatePaymentSettings(): void
+    {
+        if (static::XML_CONFIG_ENABLED === '') {
+            $this->throwValidateException('No XML config enabled path set for payment');
+        }
+
+        if (static::XML_CONFIG_ORDER_STATUS === '') {
+            $this->throwValidateException('No XML config order status set for payment');
+        }
+
+        if (static::MODULE_NAME === '') {
+            $this->throwValidateException('No module name set for payment');
+        }
+
+        if (static::PAYMENT_CODE === '') {
+            $this->throwValidateException('No payment code set for payment');
+        }
+    }
+
+    /**
+     * Throw a validate exception
+     *
+     * @param $message
+     *
+     * @throws ShopgateLibraryException
+     */
+    private function throwValidateException($message): void
+    {
+        throw new ShopgateLibraryException(ShopgateLibraryException::UNKNOWN_ERROR_CODE,
```

Comment:
chop it down, you did not install phpcs fixer as was advised today, did you?

Reply:
works now